File: java/nec.unit-test/src/test/java/nec/utils/FakeThrowsExceptionEngine.java
package nec.utils;

import nec.Models.IEngine;

public class FakeThrowsExceptionEngine implements IEngine {

    private Exception _exceptionToThrow;

    public FakeThrowsExceptionEngine(Exception exceptionToThrow) {
	this._exceptionToThrow = exceptionToThrow;
    }

    public void ignite() throws Exception {

	throw _exceptionToThrow;
    }

    public void stop() throws Exception {

	throw _exceptionToThrow;
    }

}

File: netCore/NEC.UnitTest/tests/NEC.Models.Tests/Utils/FakeThrowsExceptionEngine.cs
using System;

namespace NEC.Models.Tests.Utils
{
    public class FakeThrowsExceptionEngine : IEngine
    {
        private readonly Exception _exceptionToThrow;

        public FakeThrowsExceptionEngine(Exception exceptionToThrow) =>
            _exceptionToThrow = exceptionToThrow;

        public void Ignite() => throw _exceptionToThrow;

        public void Stop() => throw _exceptionToThrow;
    }
}
